sycl/include/CL/sycl/handler.hpp

```diff
@@ -806,12 +855,11 @@ class __SYCL_EXPORT handler {
       if (getenv("SYCL_PARALLEL_FOR_RANGE_ROUNDING_TRACE") != nullptr)
         std::cout << "parallel_for range adjusted from " << NumWorkItems[0]
                   << " to " << NewValX << std::endl;
-      auto Wrapper = [=](TransformedArgType Arg) {
-        if (Arg[0] >= NumWorkItems[0])
-          return;
-        Arg.set_allowed_range(NumWorkItems);
-        KernelFunc(Arg);
-      };
+
+      // TODO: replace detail::RangeRoundedLambda with
+      // "constexpr if" when DPC++ RT switched to C++17
+      detail::RangeRoundedLambda<TransformedArgType, KernelType, Dims> Wrapper{
```

Comment:
@rdeodhar, please review from range rounding point-of-view

Reply:
The function containing the range rounding code, "void parallel_for_lambda_impl(range<Dims> NumWorkItems, KernelType KernelFunc) {...}" is called with a KernelFunc.

KernelFunc may have a kernel_handler parameter. If it does, the wrapped kernel should be declared with that additional parameter, (say, KH), and the body of the wrapper should pass it to the original kernel, as KernelFunc(Arg, KH);

I'm not sure how you'll do it.